File: examples/eAPI v1.8/java/eAPI-v18-connector/src/main/java/cz/monetplus/mips/eapi/v18/connector/entity/ext/TrxDatesExtension.java
package cz.monetplus.mips.eapi.v18.connector.entity.ext;

import javax.xml.bind.annotation.XmlAccessType;
import javax.xml.bind.annotation.XmlAccessorType;
import javax.xml.bind.annotation.XmlElement;
import javax.xml.bind.annotation.XmlRootElement;

@XmlRootElement
@XmlAccessorType(XmlAccessType.FIELD)
public class TrxDatesExtension extends Extension {

	private static final long serialVersionUID = -3825192932302805075L;
	
	@XmlElement
	public String createdDate;
	
	@XmlElement
	public String authDate;

	@XmlElement
	public String settlementDate;
	
	@Override
	public String toSign() {
		StringBuilder sb = new StringBuilder(); 
		add(sb, extension);
		add(sb, dttm);
		add(sb, createdDate);
		add(sb, authDate);
		add(sb, settlementDate);
		deleteLast(sb);
		return sb.toString();
	}

}

File: examples/eApi v1.9/dotNET/CsobGTWDemo/CsobGatewayClientExample/Communication/DataObjects/Ext/TrxDatesExtension.cs
using System.Text;
using System.Xml.Serialization;
using Newtonsoft.Json;

namespace CsobGatewayClientExample.Communication.DataObjects.Ext;

//[XmlType("NewTypeName")]
//@XmlRootElement
//@XmlAccessorType(XmlAccessType.FIELD)
[XmlRoot("trxDates")]
public class TrxDatesExtension : Extension
{
    //private static final long serialVersionUID = -3825192932302805075L;

    //@XmlElement
    [XmlElement(ElementName = "createdDate")]
    [JsonProperty("createdDate")]
    public string? CreatedDate { get; set; }

    //@XmlElement
    [XmlElement(ElementName = "authDate")]
    [JsonProperty("authDate")]

    public string? AuthDate { get; set; }

    //@XmlElement
    [XmlElement(ElementName = "settlementDate")]
    [JsonProperty("settlementDate")]

    public string? SettlementDate { get; set; }

    //@Override
    public override string ToSign()
    {
        var sb = new StringBuilder();
        Add(sb, extension);
        Add(sb, Dttm);
        Add(sb, CreatedDate);
        Add(sb, AuthDate);
        Add(sb, SettlementDate);
        DeleteLast(sb);
        return sb.ToString();
    }
}